src/Microsoft.DotNet.Build.Tasks.Packaging/src/HarvestPackage.cs

```diff
@@ -324,22 +324,49 @@ private bool ShouldSuppress(string packagePath)
         {
             if (_pathsToSuppress == null)
             {
-                _pathsToSuppress = PathsToSuppress.NullAsEmpty().Select(EnsureTrailingSlash).ToArray();
+                _pathsToSuppress = PathsToSuppress.NullAsEmpty().Select(EnsureDirectory).ToArray();
             }
 
             return _pathsToSuppress.Any(p => packagePath.StartsWith(p, StringComparison.OrdinalIgnoreCase));
         }
 
-        private static string EnsureTrailingSlash(string source)
+        private static char[] s_slashes = new[] { '\\', '/' };
+        private static string EnsureDirectory(string source)
         {
+            string result;
+
             if (source.Length < 1 || source[source.Length - 1] == '\\' || source[source.Length - 1] == '/')
             {
-                return source;
+                // already have a directory
+                result = source;
             }
             else
             {
-                return source + '/';
+                // could be a directory of file
+                string extension = Path.GetExtension(source);
+
+                if (extension != null && extension.Length > 0 && includedExtensions.Contains(extension, StringComparer.OrdinalIgnoreCase))
+                {
+                    // it's a file, find the directory portion
+                    int lastSlash = source.LastIndexOfAny(s_slashes);
```

Comment:
can you use Path.GetFileName to check if the path ends with the file name and in same time you'll get the file name length which you can use to chop off the original path string? so you'll not need to change / to \ at that time

---
In reply to: [76501340](https://github.com/dotnet/buildtools/pull/991#discussion_r76501340) [](ancestors = 76501340)